doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -565,28 +550,32 @@ the conversion in hypotheses :n:`{+ @ident}`.
    information about the constants it encounters and the unfolding decisions it
    makes.
 
-.. tacn:: red
-   :name: red
+.. tacn:: red {? @occurrences }
 
-   This tactic applies to a goal that has the form::
+   :math:`\beta`:math:`\iota`:math:`\zeta`-reduces the head of `T`, if possible,
+   in the selected hypotheses and/or the goal, which must have the form:
 
-     forall (x:T1) ... (xk:Tk), T
+     :n:`{? forall @open_binders,} T`
 
-   with :g:`T` :math:`\beta`:math:`\iota`:math:`\zeta`-reducing to :g:`c t`:sub:`1` :g:`... t`:sub:`n` and :g:`c` a
-   constant. If :g:`c` is transparent then it replaces :g:`c` with its
-   definition (say :g:`t`) and then reduces
-   :g:`(t t`:sub:`1` :g:`... t`:sub:`n` :g:`)` according to :math:`\beta`:math:`\iota`:math:`\zeta`-reduction rules.
+   (where `T` does not begin with a `forall`) to :n:`c t__1 … t__n`
+   where :g:`c` is a constant.
+   If :g:`c` is transparent then it replaces :g:`c` with its
+   definition and reduces again until no further reduction is possible.
 
-.. exn:: Not reducible.
-   :undocumented:
+   The `at` clause of :n:`@occurrences` is not supported.
```

Comment:
Yes, but why noting it for this tactic in particular when it is actually true for all `@occurrences` clause in this section?

Reply:
It's valid for `unfold`.  I would mention it for all of them, but I'll do `simple_occurrences` instead.

```
Goal 2+3+4 = 0.
unfold Nat.add in |- * at 2.
(* 5 + 4 = 0 *)
```